lib/chef/provider/service/openbsd_rcctl.rb

```diff
@@ -0,0 +1,157 @@
+#
+# Author:: Scott Bonds (<scott@ggr.com>)
+# Author:: Joe Miller (<joeym@joeym.net>)
+# Copyright:: Copyright (c) 2014 Scott Bonds
+# Copyright:: Copyright (c) 2015 Joe Miller
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'chef/mixin/command'
+require 'chef/mixin/shell_out'
+require 'chef/provider/service/init'
+require 'chef/resource/service'
+
+class Chef
+  class Provider
+    class Service
+      class OpenbsdRcctl < Chef::Provider::Service::Init
+
+        provides :service, platform_family: 'openbsd', override: true do |node|
+          Chef::Platform::ServiceHelpers.service_resource_providers.include?(:rcctl)
+        end
+
+        include Chef::Mixin::ShellOut
+
+        def initialize(new_resource, run_context)
+          super
+          new_resource.supports[:status] = true
+          new_resource.status_command("rcctl check #{new_resource.service_name}")
+          new_resource.start_command("rcctl start #{new_resource.service_name}")
+          new_resource.stop_command("rcctl stop #{new_resource.service_name}")
+          new_resource.restart_command("rcctl restart #{new_resource.service_name}")
+          new_resource.reload_command("rcctl reload #{new_resource.service_name}")
```

Comment:
This shouldn't be mutating the new_resource and not allowing the user any chance to override these.

The supports one should be using the superclass method `#supports` which dups a copy of the supports hash in the new_resource.  It should then be more like `supports[:status] = false if supports[:status].nil?` so that the user can set true/false and override.

you should also take a similar approach with the commands where they should not mutate the new resource but should be ivars on the provider -- and those values you have should be defaults and the new_resource should be able to override them.